### PR TITLE
PullRequest for Issue126: Refactor the dwell implemention for Amptek 

### PR DESCRIPTION
**Pull request by XuetaoLiu-S**
https://github.com/acquaman/AcquamanDataServer/issues/126

### CODE REVIEW

**Comment by XuetaoLiu-S** [comment]
@davidChevrier 
I removed the commented codes and make some small changes. Pls take a look.

**Comment by XuetaoLiu-S** [comment]
Tested OK

**Comment by dretrex** [comment]
Looks okay to me.  @davidChevrier should probably still go over it as well.
Time Estimate: 5 minutes

File: source/Detector/Amptek/SGM/AmptekSDD123EPICSDetectorManager.h [hunk 1]
```diff
@@ -63,11 +63,11 @@ protected slots:
 
 protected:
 	/// helper function to handle continuous spectrum data update
-	void dataHelper(AMDSDataHolder *spectrum, int count, double elapsedTime);
+	void dataHelper(AMDSDataHolder *spectrum, double count, double elapsedTime);
```
Comment (davidChevrier) [inline source/Detector/Amptek/SGM/AmptekSDD123EPICSDetectorManager.h]: Why did the count parameter change to double here?

**Comment by davidChevrier** [comment]
Overall, it looks very good to me. Thanks for catching a bunch of issues @XuetaoLiu-CLS 
I think this is ready for testing, but we'll have to wait to make sure that SGM comes up in a stable condition before actually doing these tests.

Time Estimate: 20 minutes

**Comment by XuetaoLiu-S** [comment]
@davidChevrier I tested it yesterday, the step scan works as expected.

File: source/Detector/Amptek/SGM/AmptekSDD123EPICSDetectorManager.h (hunk 1 above)
Comment (XuetaoLiu-S) [inline source/Detector/Amptek/SGM/AmptekSDD123EPICSDetectorManager.h]: There is a compile warning. in the following codes, we passed in a double.

```
void AmptekSDD123EPICSDetectorManager::setFlattenedSpectrumData(AMDSDataHolder *dataHolder)
  {
    connect(spectrumControl_, SIGNAL(valueChanged(double)), this, SLOT(onSpectrumControlValueChanged()));

    double elapsedTime = dwellTime();
    double count = elapsedTime*1000/50;
    dataHelper(dataHolder, count, elapsedTime);
  }
```

Comment (dretrex) [inline source/Detector/Amptek/SGM/AmptekSDD123EPICSDetectorManager.h]: Isn't count the number of data positions?  That should definitely be an int if that's the case.

**Comment by davidChevrier** [comment]
@iainworkman 
I'm about to merge this into master. Is there anything we need to be cognizant of when we want get this in as a release candidate?

**Comment by iainworkman** [comment]
No, I will bear it in mind when I'm doing the deploy. It'll affect the test procedures which I perform (ensuring the dwells work as expected, and the step scanning is okay).

**Comment by XuetaoLiu-S** [comment]
@iainworkman we can deploy after this and #129 together so that we can deploy the AMDSPV at the same time

**Comment by davidChevrier** [comment]
@iainworkman @XuetaoLiu-CLS 
I think it would be fairly reasonable to wait to deploy this with #129 
I think that SGM needs some time and user shifts to demonstrate they are happy with the current deployment before we go again. @iainworkman could you check into the user schedule and see what the next 2-3 weeks look like?

**Pull request merged**